Contact Us Feature Layout

colinwilliams91 opened the pull request:
- z-index clarified ✔️ 
- moved common `styles/colors` reference into DEVELOPMENT.md in root ✔️ 
- `mailto` pending ❌ 
- cleaned up unused imports/comments ✔️ 
- 'lorem ipsum' copy placeholder waiting for Jenna or Deacon ❌ 
- resolved `App.tsx` conflict ✔️ 
- resolved `package-lock.json` conflict ✔️ 

Contact Feature is basically complete. App is functioning correctly. Still needs to be wired to backend.

colinwilliams91 committed:
link socials urls

colinwilliams91 committed:
center send photo button, hope it is desired

colinwilliams91 committed:
update instagram url

colinwilliams91 committed:
update readme social links

colinwilliams91 committed:
set up route and path to new contact-us component

colinwilliams91 committed:
logo navigates to home from other views now

colinwilliams91 committed:
text area, two buttons, styled contact, logo routes home accross views

colinwilliams91 committed:
re-styling for wireframes

colinwilliams91 committed:
compose contact feature dir & components, view layout

colinwilliams91 committed:
Merge pull request #1 from colinwilliams91/contact-v1.0

Contact v1.0

colinwilliams91 committed:
README update

colinwilliams91 committed:
Merge pull request #2 from colinwilliams91/contact-v1.0

README update

colinwilliams91 committed:
add message, add faq, create new button

colinwilliams91 committed:
Merge branch 'main' of github.com:colinwilliams91/qg-frontend-v2 into contact-v1.1
update origin from org, new contact 3 new panels

colinwilliams91 committed:
Merge pull request #3 from colinwilliams91/contact-v1.1

Contact v1.1

colinwilliams91 committed:
Reach us and Volunteer done

colinwilliams91 committed:
contact message section filled out

colinwilliams91 committed:
merge home change 5/14

colinwilliams91 committed:
no un-used vars fix/comment

colinwilliams91 committed:
Merge pull request #4 from colinwilliams91/contact-v1.3

big updates to home and navs, my contact feature 80% complete

colinwilliams91 committed:
transfered my pvt clone to this

colinwilliams91 committed:
href=/ topNavLogo to route home from all views

colinwilliams91 committed:
comment for unused code

colinwilliams91 committed:
container message

colinwilliams91 committed:
npm i latest react-bootstrap, contact layout done

colinwilliams91 committed:
remove unused, move common colors to dev.md

colinwilliams91 committed:
rm rf npm i

colinwilliams91 committed:
merge upstream into local main (npm i current)

colinwilliams91 committed:
manually merged pkg-lock, pull = fallback main now

colinwilliams91 committed:
upstream package-lock

colinwilliams91 committed:
escaped apos contactFAQ

mnyon-grandkru commented on DEVELOPMENT.md:
````diff
@@ -0,0 +1,29 @@
+# Common Color Hex Codes
+
+```css
````
Love that you've added this to documentation

Githubbubber replied:
Me too. Makes sense!

mnyon-grandkru commented on README.md:
```diff
@@ -26,7 +26,7 @@ Site Milestone Dates\*: ~~June 30th~~✓ | August 31st | October 11th
   --- Logout (links to blank page)
 - Support
-  --- Contact Us (links to blank page)
+  --- Contact Us (links to skeleton)
```
Good catch. Is there a Contact Us page in the mockups? I can't find it, but I'm looking for something labeled "Contact Us"

colinwilliams91 replied:
I've been following page 32 out of 125 [here](https://xd.adobe.com/view/6f6af599-8eb6-46b5-8731-e4c09ddd587e-3a39/screen/8875368e-2396-45c9-8ff1-b53624f6270f/) from the mockups

mnyon-grandkru commented on package.json:
```diff
@@ -13,9 +13,9 @@
     "@types/node": "^16.11.1",
     "@types/react": "^17.0.30",
     "@types/react-dom": "^17.0.9",
-    "bootstrap": "^5.1.1",
```
This will settle some security upgrade issues.

mnyon-grandkru commented on src/components/common/footer/Footer.tsx:
```diff
@@ -1,42 +1,44 @@
 import SocialIcons from "../social-icons/SocialIcons";
 import icon from "../../../assets/footer/QG-Logo-V3-Gold-Transparent-PNG-1.png";
 
+// import Contact from "../../contact/Contact";
```
Can we remove this if it's not being used?

Githubbubber replied:
+1 on this one. Will remember to not do this, myself

mnyon-grandkru commented on src/components/common/topnav/TopNav.css.ts:
```diff
@@ -32,6 +36,7 @@ export const NavLink = styled(Nav.Link)`
 export const TopNavLogo = styled.span`
 	text-align: left;
 	flex-grow: 1;
+	z-index: 50;
```
I'd asked about this before. I'm assuming this is where things panned out?

colinwilliams91 replied:
yup, I have seen a couple approaches to using `z-index` and I'm not sure what the best practice is. Maybe we should use 1, 2, 3... etc. To be on the safe side I put it really high (50) so it would be on top of everything but I'll look around and see if there is anything else using it. Do you have an feelings one way or the other for that?

mnyon-grandkru replied:
I don't, so let it be.

mnyon-grandkru commented on src/components/contact/Contact.tsx:
```diff
@@ -0,0 +1,36 @@
+import {
+  ContactWrapper,
+  HeaderPurpleTransparentLeft,
+  // HeaderPurpleTransparentTop, // <-- plan on using this
```
Ah, will this be used in this ticket?

mnyon-grandkru commented on src/components/contact/contactCTA/ContactCTA.css.ts:
```diff
@@ -0,0 +1,71 @@
+
+/*
+   ***************************************
+  *************** PANELS ******************
```
Love this!

colinwilliams91 replied:
😊 😊 😊

mnyon-grandkru commented on src/components/contact/contactCTA/ContactCTA.tsx:
```diff
@@ -0,0 +1,72 @@
+  return (
+    <PanelsContainer id="panels-box">
+      <WhiteBGSectionLeft id="white-bg-section">
+        <TitleText>REACH US</TitleText>
```
How would you feel about taking content in tags like `<TitleText>` and `<ReachSpacer>` amongst others, and placing it in a `Contact.onstants.ts` file? As we make more of these, we can refactor those constants up to `App.contstants.ts`

colinwilliams91 replied:
I think that's a great idea. So, if I understand correctly, make some more generic components that value can be plugged into for different use cases?

mnyon-grandkru replied:
Not full fledged components, but a JS file to hold constants. It's similar to techniques used to handle internationalization, but we don't need to do it for multiple languages. Just put string constants into a separate file and import it into `Contact.tsx`

mnyon-grandkru commented on src/components/contact/contactFAQ/ContactFAQ.tsx:
```diff
@@ -0,0 +1,134 @@
+              <b>How do I add a resource?</b>
+            </Accordion.Header>
+            <Accordion.Body>
+              Elit dolor irure dolore reprehenderit Lorem ad id culpa nulla
```
Do we have copy from the mockups or others that can be substituted here. Let's get rid of the Latin.

colinwilliams91 replied:
Yeah I agree, I used all the copy available in the mockups. Not sure where else to look..

mnyon-grandkru replied:
I think we can close this for now.

Githubbubber replied:
For sure we need a check-in regarding the copy. The dang VOLUNTEER header for the contact us page is “VOLUNTEERING” for the mobile view. :/

mnyon-grandkru commented on src/components/contact/contactFAQ/ContactFAQ.tsx:
```diff
@@ -0,0 +1,134 @@
+                <li>Edit your username or email address as desired.</li>
+              </ol>
+              <p>
+                NOTE: updating your username will require confirmation by email.
```
Is this logic that we'll need to implement at a later date?

colinwilliams91 replied:
I'm not actually sure about the logic, this just came directly from the mockups (I'm not sure of its origin and if it is actually agnostic to previous logic or not)

mnyon-grandkru replied:
Yeah, we can confirm if that's desired behavior at a future meeting.

mnyon-grandkru commented on src/components/contact/contactMessage/ContactMessage.tsx:
```diff
@@ -0,0 +1,38 @@
+          <span>
+            <b>Email</b>
+          </span>
+          <input type="text" style={{ border: "1px solid #B5B5B5" }}></input>
```
Is there a reason why these styles are inline?

colinwilliams91 replied:
nope! I'll put it into my "modularize" refactor 😄

colinwilliams91 committed:
fix contact header, changed message inputs

colinwilliams91 commented:
- fixed contact us use of
- moved inline styling to re-usable TextInput and SelectInput styled components
- added dropdown for rendered Message component

Githubbubber commented on src/components/common/topnav/TopNav.tsx:
```diff
@@ -1,51 +1,56 @@
+  );
+};
+
+export default TopNav;
```
Not sure if there will be an error around there not being an empty new line. Hope not.

Githubbubber commented on src/components/contact/Contact.tsx:
```diff
@@ -0,0 +1,37 @@
+        <Title>CONTACT US</Title>
+      </TitleWrapper>
+      <ContactWrapper id="main">
+        <ContactMain />
```
This has the “VOLUNTEER" section in it, yes? No need for the "REACH US” section then, unless there’s a design I’m not finding on my end?

Mekebrown reviewed (approved):
LGTM. I understand all the work and know that it’s a WIP. But even so it’s good to merge for a workable Contact Us desktop page, IMO.

colinwilliams91 commented:
I deleted my fork at [colinwilliams91/qg-frontend-v2](https://github.com/colinwilliams91/qg-frontend-v2) and merged all of this plus Meke's fix #67 into main (pulled into my new branch) as well as the `react-scripts` v5 bump. Will open a new branch and PR since I can't seem to push since this PR came from a different Fork
